### PR TITLE
[IMP] mail: turn "Chatter/isPreparingAttachmentsLoading" into field

### PR DESCRIPTION
**Pull request by alexkuhn**
Task-2734101

### CODE REVIEW

File: addons/mail/static/src/models/chatter/chatter.js [hunk 1]
```diff
@@ -340,6 +341,9 @@ registerModel({
             compute: '_computeIsDisabled',
             default: false,
         }),
+        isPreparingAttachmentsLoading: attr({
```
Comment (seb-odoo) [inline addons/mail/static/src/models/chatter/chatter.js:344]: I think the field is actually useless, it's equivalent to `attachmentsLoaderTimeout !== undefined`?

File: addons/mail/static/src/models/chatter/chatter.js [hunk 2]
```diff
@@ -176,8 +173,10 @@ registerModel({
          * @private
          */
         _onAttachmentsLoadingTimeout() {
-            this.update({ isShowingAttachmentsLoading: true });
-            this._isPreparingAttachmentsLoading = false;
+            this.update({
+                isPreparingAttachmentsLoading: clear(),
+                isShowingAttachmentsLoading: true,
+            });
```
Comment (seb-odoo) [inline addons/mail/static/src/models/chatter/chatter.js:179]: Looks like we forgot to clear `attachmentsLoaderTimeout` (even before this PR).

File: addons/mail/static/src/models/chatter/chatter.js [hunk 3]
```diff
@@ -176,8 +173,11 @@ registerModel({
          * @private
          */
         _onAttachmentsLoadingTimeout() {
-            this.update({ isShowingAttachmentsLoading: true });
-            this._isPreparingAttachmentsLoading = false;
+            this.messaging.browser.clearTimeout(this.attachmentsLoaderTimeout);
```
Comment (seb-odoo) [inline addons/mail/static/src/models/chatter/chatter.js:176]: I meant to `clear()` the timeout holding field, not clear the timeout itself.

File: addons/mail/static/src/models/chatter/chatter.js (hunk 1 above)
Comment (seb-odoo) [reply addons/mail/static/src/models/chatter/chatter.js:345]: Comment not done, can we look the not introduce this extra field?

Comment (alexkuhn) [reply addons/mail/static/src/models/chatter/chatter.js:347]: Done, turned into computed field.

Comment (seb-odoo) [reply addons/mail/static/src/models/chatter/chatter.js:347]: Ok, it still looks like a useless indirection, but it's already much better with a compute.

**Review by seb-odoo — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**